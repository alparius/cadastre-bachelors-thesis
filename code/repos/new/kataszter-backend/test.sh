#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
