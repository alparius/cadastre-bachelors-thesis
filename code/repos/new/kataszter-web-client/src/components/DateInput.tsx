import React, { useState } from "react";
import { FormattedMessage as FM } from "react-intl";
import { Form, Message } from "semantic-ui-react";

import IDate from "../data/Date";
import checkDate from "../util/dateConverts/checkDate";
import BoldFM from "../util/styledMessages/BoldFM";
import ItalicFM from "../util/styledMessages/ItalicFM";

const regexp = /^[0-9\b]+$/;

interface IDateInputProps {
    date: IDate;
    isGrey?: boolean;
    changeDate(newDate: IDate): void;
}

const DateInput: React.FC<IDateInputProps> = (props: IDateInputProps) => {
    const { date, changeDate, isGrey } = props;

    const [dateIsValid, setDateIsValid] = useState(checkDate(date));

    const handleChange = (event: any) => {
        const { name, value } = event.target;

        if (name === "comment") {
            changeDate({ ...date, comment: value });
        } else if (value === "" || regexp.test(value)) {
            setDateIsValid(checkDate({ ...date, [name]: Number(value) }));
            changeDate({ ...date, [name]: Number(value) });
        }
    };

    return (
        <Form>
            <Form.Group inline unstackable style={{ marginBottom: "-2px" }}>
                <Form.Input
                    style={{ width: "6em" }}
                    fluid
                    id={isGrey ? "grey" : ""}
                    label={<BoldFM id="label.date.year" style={isGrey ? { color: "gray" } : {}} />}
                    type="number"
                    name="year"
                    value={date.year === 0 ? "" : date.year}
                    onChange={handleChange}
                />
                <Form.Input
                    style={{ width: "5em" }}
                    fluid
                    id={isGrey ? "grey" : ""}
                    label={<BoldFM id="label.date.month" style={isGrey ? { color: "gray" } : {}} />}
                    type="number"
                    name="month"
                    value={date.month === 0 ? "" : date.month}
                    onChange={handleChange}
                />
                <Form.Input
                    style={{ width: "5em" }}
                    fluid
                    id={isGrey ? "grey" : ""}
                    label={<BoldFM id="label.date.day" style={isGrey ? { color: "gray" } : {}} />}
                    type="number"
                    name="day"
                    value={date.day === 0 ? "" : date.day}
                    onChange={handleChange}
                />
                {!dateIsValid && (
                    <Form.Field
                        label={
                            <Message negative size="mini">
                                <FM id="formError.invalidDate" />
                            </Message>
                        }
                    />
                )}
            </Form.Group>
            <Form.Input
                fluid
                id={"grey"}
                label={<ItalicFM id="label.date.comment" style={{ color: "gray" }} />}
                name="comment"
                value={date.comment}
                onChange={handleChange}
            />
        </Form>
    );
};

export default DateInput;
