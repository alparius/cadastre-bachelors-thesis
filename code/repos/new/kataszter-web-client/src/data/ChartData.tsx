interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
