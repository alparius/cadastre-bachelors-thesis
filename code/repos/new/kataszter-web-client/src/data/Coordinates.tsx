interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
