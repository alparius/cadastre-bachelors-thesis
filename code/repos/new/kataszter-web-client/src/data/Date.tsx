interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
