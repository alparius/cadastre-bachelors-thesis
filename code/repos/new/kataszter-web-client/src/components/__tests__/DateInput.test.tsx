/*===================
  DateInput.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestDate } from "../../util/test/data/test-data";
import { createComponentWithIntl, mountWithIntl } from "../../util/test/test-wrapper";
import DateInput from "../DateInput";

const changeDate = jest.fn();

describe("DateInput element", () => {
    afterEach(() => {
        changeDate.mockClear();
    });

    it("should render a form", () => {
        const container = createComponentWithIntl(<DateInput date={NewTestDate} changeDate={jest.fn()} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message too on bad input", () => {
        const container = mountWithIntl(<DateInput date={NewTestDate} changeDate={jest.fn()} />);
        const input = container.find("input").at(1);
        input.simulate("change", { target: { value: 13 } });
        expect(container).toMatchSnapshot();
    });

    it.each([
        [0, "2", "3"],
        [1, "2", "3"],
        [2, "2", "3"],
        [3, "2", "3"]
    ])("should  call changeDate on %d. input. After first change: %s, After second change: %s", (inputNr, firstChange, secondChange) => {
        const container = mountWithIntl(<DateInput date={NewTestDate} changeDate={changeDate} />);
        const input = container.find("input").at(Number(inputNr));
        input.simulate("change", { target: { value: firstChange } });
        input.simulate("change", { target: { value: secondChange } });
        expect(changeDate).toHaveBeenCalledTimes(2);
    });
});
