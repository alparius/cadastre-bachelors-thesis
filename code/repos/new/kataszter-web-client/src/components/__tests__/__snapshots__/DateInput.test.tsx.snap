// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput element should render a form 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline unstackable fields"
    style={
      Object {
        "marginBottom": "-2px",
      }
    }
  >
    <div
      className="field"
    >
      <b
        style={Object {}}
      >
        year
      </b>
      <div
        className="ui fluid input"
        style={
          Object {
            "width": "6em",
          }
        }
      >
        <input
          id=""
          name="year"
          onChange={[Function]}
          type="number"
          value={2000}
        />
      </div>
    </div>
    <div
      className="field"
    >
      <b
        style={Object {}}
      >
        month
      </b>
      <div
        className="ui fluid input"
        style={
          Object {
            "width": "5em",
          }
        }
      >
        <input
          id=""
          name="month"
          onChange={[Function]}
          type="number"
          value={6}
        />
      </div>
    </div>
    <div
      className="field"
    >
      <b
        style={Object {}}
      >
        day
      </b>
      <div
        className="ui fluid input"
        style={
          Object {
            "width": "5em",
          }
        }
      >
        <input
          id=""
          name="day"
          onChange={[Function]}
          type="number"
          value={6}
        />
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <i
      style={
        Object {
          "color": "gray",
        }
      }
    >
      comment about date (optional)
    </i>
    <div
      className="ui fluid input"
    >
      <input
        id="grey"
        name="comment"
        onChange={[Function]}
        type="text"
        value="2000.06.06"
      />
    </div>
  </div>
</form>
`;

exports[`DateInput element should render an error message too on bad input 1`] = `ReactWrapper {}`;
